### PR TITLE
Moar tests

### LINKED ISSUE #8: test stream filter behaviors
@pietgeursen, if we wanna test all the things... here's some moar i realized:

- [x] [`models()` stream won't receive a new item when it's the same as the last item](https://github.com/ahdinosaur/inu/blob/master/start.js#L39)
- [x] [`views()` stream won't receive `null` or `undefined` views](https://github.com/ahdinosaur/inu/blob/master/start.js#L49)
- [x] [`effects()` stream won't receive `null` or `undefined` effects](https://github.com/ahdinosaur/inu/blob/master/start.js#L59)
- [x] [`effectActionStreams()` won't receive `null` or `undefined` streams returned from `run`](https://github.com/ahdinosaur/inu/blob/master/start.js#L69)

but i'm also quite happy that we have any tests at all. :blush:

### PR DESCRIPTION
**Pull request by pietgeursen**
Resolves #8 
I'm coding while sick and a bit foggy so expect there to be typos / small mistakes... :)

### CODE REVIEW

File: test/update.js [hunk 1]
```diff
@@ -27,3 +27,75 @@ test('models returned by update are emitted by the model stream', function (t) {
...
+    },
+    update: function (model, action) {
+      t.ok(model)
+      return {model, effect: null}
+    },
+    view: function (model, dispatch) {
+      dispatch()
+    },
+    run: function (effect) {
+      t.error(effect)
```
Comment (ahdinosaur) [inline test/update.js]: [i reckon we should use `t.fail` here](https://github.com/substack/tape#tfailmsg)

File: test/update.js [hunk 2]
```diff
@@ -27,3 +27,75 @@ test('models returned by update are emitted by the model stream', function (t) {
...
+    },
+    update: function (model, action) {
+      t.ok(model)
+      return {model, effect: undefined}
+    },
+    view: function (model, dispatch) {
+      dispatch()
+    },
+    run: function (effect) {
+      t.error(effect)
```
Comment (ahdinosaur) [inline test/update.js]: more `t.fail`

File: test/update.js [hunk 3]
```diff
@@ -27,3 +27,75 @@ test('models returned by update are emitted by the model stream', function (t) {
...
+    }
+  }
+  var sources = inu.start(app)
+
+  pull(sources.effects(), pull.drain(function (model) {
+    t.error(true)
```
Comment (ahdinosaur) [inline test/update.js]: more `t.fail`

File: test/update.js [hunk 4]
```diff
@@ -27,3 +27,75 @@ test('models returned by update are emitted by the model stream', function (t) {
...
+    view: function (model, dispatch) {
+      dispatch()
+    },
+    run: function (effect) {
+      t.error(effect)
+    }
+  }
+  var sources = inu.start(app)
+
+  pull(sources.effects(), pull.drain(function (model) {
```
Comment (ahdinosaur) [inline test/update.js]: more `t.fail`

File: test/run.js [hunk 5]
```diff
@@ -134,3 +134,25 @@ test('actions stream passed to run emits actions', function (t) {
...
+    },
+    view: function (model, dispatch) {},
+    run: function (effect) {
+      t.ok(effect)
+    }
+  }
+  var sources = inu.start(app)
+
+  pull(sources.effectActionsSources(), pull.drain(function (model) {
+    t.error(true)
```
Comment (ahdinosaur) [inline test/run.js]: more `t.fail`

**Comment by ahdinosaur** [comment]
wow this is great, thanks heaps @pietgeursen. :smiley_cat: left some comments where we should probably use [`t.fail`](https://github.com/substack/tape#tfailmsg) instead of `t.error`, otherwise looks good to me..

hope you feel better eyy. :lemon: :tea: :honey_pot:

**Comment by pietgeursen** [comment]
Thanks for the t.fail hint @ahdinosaur . Hadn't seen it before and didn't think to go look at the docs.

**Comment by yoshuawuyts** [comment]
`t.fail` doesn't add to line coverage tho - which in practical sense is irrelevant but does report like missed lines in codecov 😔

**Comment by ahdinosaur** [comment]
@yoshuawuyts hmm, so is it better to do `t.ok(false)`, or ...?

**Comment by yoshuawuyts** [comment]
> hmm, so is it better to do t.ok(false), or ...?

Hah yeah so I tend to not test those branches anymore; not saying that that's the right solution but... idk - there's something to be said for either approach. I enjoy the idea of having touched every single code branch with tests, and `t.fail()` makes it so that it never happens. Sorry for barging in only to provide a non-answer haha

**Pull request merged**